server/grpc_service.go

```diff
@@ -290,27 +295,13 @@ func (s *Server) RegionHeartbeat(server pdpb.PD_RegionHeartbeatServer) error {
 			continue
 		}
 
-		var resp *pdpb.RegionHeartbeatResponse
-		resp, err = cluster.handleRegionHeartbeat(region)
+		err = cluster.handleRegionHeartbeat(region)
 		if err != nil {
 			msg := errors.Trace(err).Error()
 			err = sendErrorRegionHeartbeatResponse(server, s.clusterID, pdpb.ErrorType_UNKNOWN, msg)
```

Comment:
I think no need to drop the stream. The stream itself is in good condition, only some requests are not able to respond.